src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -335,12 +348,19 @@ internal static ClassType GetClassType(Type type)
                 type = Nullable.GetUnderlyingType(type);
             }
 
-            // A Type is considered a value if it implements IConvertible.
+            // A Type is considered a value if it implements IConvertible or is a DateTimeOffset.
             if (typeof(IConvertible).IsAssignableFrom(type) || type == typeof(DateTimeOffset))
             {
                 return ClassType.Value;
             }
 
+            if (typeof(IDictionary).IsAssignableFrom(type) || 
+                (type.IsGenericType && (type.GetGenericTypeDefinition() == typeof(IDictionary<,>) 
```

Comment:
Store `type.GetGenericTypeDefinition()` in a local?

Reply:
It's already cached in the corresponding method.

Reply:
In this **local** branch condition, we call `type.GetGenericTypeDefinition()` twice.

What is the corresponding method in which it's cached and how does it help here?